packages/util/src/bytes.ts

```diff
@@ -25,19 +25,19 @@ export const bytesToUnprefixedHex = _bytesToUnprefixedHex
  * @throws If the input is not a valid 0x-prefixed hex string
  */
 export const hexToBytes = (hex: string) => {
-  if (!hex.startsWith('0x')) throw EthereumJSErrorWithoutCode('input string must be 0x prefixed')
+  if (!isHexString(hex)) throw EthereumJSErrorWithoutCode('input string must be 0x prefixed')
```

Comment:
Our `isHexString` method is rather strict, since it will also check a regex pattern to verify that the contents after `0x` are correct hex values. (So `0xaabbZZcc` will return `false` and thus throw this method).

Could you check if `nobleH2B` has this behavior internally? We should otherwise NOT change this and instead use `hex.startsWith('0x')`. The reason is that this method is called in a lot of places where performance is important and we should thus ensure that this method does not verify inputs twice.

Reply:
HexToBytes from noble does indeed do some verification (just not the 0x): https://github.com/paulmillr/noble-hashes/blob/main/src/utils.ts#L110C8-L110C16
Agree with your assessment not to duplicate verification, will revert! ty